test/core/client_channel/http_proxy_mapper_test.cc

```diff
@@ -162,43 +209,77 @@ TEST(NoProxyTest, InvalidCIDREntries) {
 }
 
-TEST(ProxyForAddressTest, ChannelArgPreferred) {
+TEST(ProxyForAddressTest, AddressesNotIncluded) {
   ScopedEnvVar address_proxy(HttpProxyMapper::kAddressProxyEnvVar,
                              "192.168.0.100:2020");
-  auto args =
-      ChannelArgs().Set(GRPC_ARG_ADDRESS_HTTP_PROXY, "192.168.0.101:2020");
-  auto address = StringToSockaddr("192.168.0.1:3333");
+  ScopedEnvVar address_proxy_enabled(
+      HttpProxyMapper::kAddressProxyEnabledAddressesEnvVar,
+      " 192.168.0.0/24 , 192.168.1.1 , 2001:db8:1::0/48 , 2001:db8:2::5");
+  // v4 address
+  auto address = StringToSockaddr("192.168.2.1:3333");
+  ChannelArgs args;
   ASSERT_TRUE(address.ok()) << address.status();
-  auto mapped = HttpProxyMapper().MapAddress(*address, &args);
-  ASSERT_TRUE(mapped.has_value());
-  auto address_string = grpc_sockaddr_to_string(&mapped.value(), true);
-  ASSERT_TRUE(address_string.ok()) << address.status();
-  EXPECT_EQ(*address_string, "192.168.0.101:2020");
-  EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), "192.168.0.1:3333");
+  EXPECT_EQ(HttpProxyMapper().MapAddress(*address, &args), absl::nullopt);
+  EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), absl::nullopt);
+  // v6 address
+  address = StringToSockaddr("[2001:db8:2::1]:3000");
+  args = ChannelArgs();
+  ASSERT_TRUE(address.ok()) << address.status();
+  EXPECT_EQ(HttpProxyMapper().MapAddress(*address, &args), absl::nullopt);
+  EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), absl::nullopt);
 }
 
 TEST(ProxyForAddressTest, BadProxy) {
   auto args = ChannelArgs().Set(GRPC_ARG_HTTP_PROXY, "192.168.0.0.100:2020");
   auto address = StringToSockaddr("192.168.0.1:3333");
   ASSERT_TRUE(address.ok()) << address.status();
-  auto mapped = HttpProxyMapper().MapAddress(*address, &args);
-  EXPECT_FALSE(mapped.has_value());
+  EXPECT_EQ(HttpProxyMapper().MapAddress(*address, &args), absl::nullopt);
   EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), absl::nullopt);
 }
+
+class IncludedAddressesTest
+    : public ::testing::TestWithParam<absl::string_view> {};
+
+INSTANTIATE_TEST_CASE_P(IncludedAddresses, IncludedAddressesTest,
+                        ::testing::Values(
+                            // IP v6 address in a proxied subnet
+                            "[2001:db8:1::1]:2020",
+                            // IP v6 address that is proxied
+                            "[2001:db8:2::5]:2020",
+                            // Proxied IP v4 address
+                            "192.168.1.1:3333",
+                            // IP v4 address in proxied subnet
+                            "192.168.0.1:3333"));
+
+TEST_P(IncludedAddressesTest, AddressIncluded) {
+  ScopedEnvVar address_proxy(HttpProxyMapper::kAddressProxyEnvVar,
+                             "[2001:db8::1111]:2020");
+  ScopedEnvVar address_proxy_enabled(
+      HttpProxyMapper::kAddressProxyEnabledAddressesEnvVar,
+      // Whitespaces are meaningful
```

Comment:
Why?  The code looks like it won't care.

Wait -- are you saying that we want the whitespace in the test to prove that the code will not care about the whitespace?  If so, please say that explicitly.

Reply:
Done